File: Native/Test/Fabric/all-operator-parameters.js
FC = createFabricClient();
FABRIC = FC.wrapFabricClient(FC);

for( var i = 1; i < 32; i++ )
{
  var klCode = "operator foo( ";
  for( var j = 0; j < i; j++ )
  {
    if( j > 0 ) klCode += ", ";
    klCode += "io Integer v" + j;
  }
  klCode += " ) { ";
  for( var j = 0; j < i; j++ )
    klCode += "v" + j + " -= " + (j+1) + "; ";
  klCode += " }";
  
  op = FABRIC.DependencyGraph.createOperator("foo_"+i);
  op.setEntryFunctionName("foo");
  op.setSourceCode(klCode);

  node = FABRIC.DependencyGraph.createNode("bar_"+i);

  var layout = [];
  for( var j = 0; j < i; j++ )
  {
    layout.push( "self.p" + j );
    node.addMember( "p" + j, "Integer" );
	node.setData( "p" + j, 0, j+1 );
  }

	binding = FABRIC.DG.createBinding();
	binding.setOperator(op);
	binding.setParameterLayout(layout);
  
  node.bindings.append(binding);
  node.evaluate();
  
  for( var j = 0; j < i; j++ )
  {
    if( node.getData( "p" + j, 0 ) != 0 )
      print( "Parameter " + j + " fails when evaluating " + i + " params" );
  }
}

FABRIC.flush();
FC.dispose();

File: Native/Test/Python/all-operator-parameters.py
#
#  Copyright 2010-2012 Fabric Technologies Inc. All rights reserved.
#

import fabric
fabricClient = fabric.createClient()

for i in range( 1, 32 ):
  klCode = "operator foo( "
  for j in range( 0, i ):
    if j > 0: klCode += ", "
    klCode += "io Integer v" + str(j)
  klCode += " ) { "
  for j in range( 0, i ):
    klCode += "v" + str(j) + " -= " + str(j+1) + "; "
  klCode += " }"
  
  op = fabricClient.DependencyGraph.createOperator("foo_"+str(i))
  op.setEntryFunctionName("foo")
  op.setSourceCode(klCode)

  node = fabricClient.DependencyGraph.createNode("bar_"+str(i))

  layout = []
  for j in range( 0, i ):
    layout.append( "self.p" + str(j) )
    node.addMember( "p" + str(j), "Integer" )
    node.setData( "p" + str(j), 0, j+1 )

  binding = fabricClient.DG.createBinding()
  binding.setOperator(op)
  binding.setParameterLayout(layout)
  
  node.bindings.append(binding)
  node.evaluate()
 
  for j in range( 0, i ):
    if node.getData( "p" + str(j), 0 ) != 0:
      print( "Parameter " + str(j) + " fails when evaluating " + str(i) + " params" )

fabricClient.close()


